dukan_presentation/src/commonMain/kotlin/net/thechance/mena/dukan/presentation/viewModel/manageShelf/ManageShelfViewModel.kt

```
@@ -0,0 +1,30 @@
+package net.thechance.mena.dukan.presentation.viewModel.manageShelf
+
+import androidx.lifecycle.SavedStateHandle
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.IO
+import net.thechance.mena.dukan.presentation.screen.manageShelf.ManageShelfArgs
+import net.thechance.mena.dukan.presentation.viewModel.base.BaseViewModel
+
+class ManageShelfViewModel(
+    savedStateHandle: SavedStateHandle,
+    defaultDispatcher: CoroutineDispatcher = Dispatchers.IO
+) : BaseViewModel<ManageShelfUiState, ManageShelfEffect>(
+    initialState = ManageShelfUiState(), defaultDispatcher = defaultDispatcher
+), ManageShelfInteractionListener {
+
+    init {
+        val shelfId: String = requireNotNull(savedStateHandle[ManageShelfArgs.shelfId])
```

Maybe consider providing a default value here so it doesn’t crash if args are missing instead of using requireNotNull
what do you think?

---

i think shelf id should be required, but title i added a default value